src/services/suggestionDiagnostics.ts

```diff
@@ -104,4 +119,18 @@ namespace ts {
     function getErrorNodeFromCommonJsIndicator(commonJsModuleIndicator: Node): Node {
         return isBinaryExpression(commonJsModuleIndicator) ? commonJsModuleIndicator.left : commonJsModuleIndicator;
     }
+
+    function containsAsync(arr: NodeArray<Modifier>): boolean{
+        for(let modifier of arr){
+            if(modifier.kind === SyntaxKind.AsyncKeyword){
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    function isPromiseType(T:Type):boolean{

```

Comment:
a better check is to expose the checker's `getPromisedTypeOfPromise`, call that, if it returns a type, then it is `Promise`, if it returns `undefined` it is not.